lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -326,11 +338,27 @@ sub engine_workit {
         my $error = locate_local_assets(\%vars, $assetkeys);
         return $error if $error;
     }
+    my $casedir     = OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $productdir  = OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    my $target_name = path($casedir)->basename;
 
     $vars{ASSETDIR}   //= OpenQA::Utils::assetdir();
-    $vars{CASEDIR}    //= OpenQA::Utils::testcasedir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
-    $vars{PRODUCTDIR} //= OpenQA::Utils::productdir($vars{DISTRI}, $vars{VERSION}, $shared_cache);
+    $vars{CASEDIR}    //= $target_name;
+    $vars{PRODUCTDIR} //= substr($productdir, rindex($casedir, $target_name));
+
+    if (!is_http_url($vars{CASEDIR})) {
+        if (my $error = _link_repo($casedir, $pooldir, $target_name)) { return $error }
+    }
 
+    # if NEEDLES_DIR is an absolute path, it means that users or openqa-clone-custom-git-refspec specify it,
+    # if NEEDLES_DIR is an URL address, it means that users specify it and want to get the needles from URL.
+    # These two scenarios, we do not need to do the symlink.
```

Comment:
Can you phrase the comment in a way that it describes why we can not create links rather than "not need it"?

Reply:
modified. Please help me confirm that if the new words make sense.